src/components/productView/productViewMissing.js

```diff
@@ -27,41 +25,21 @@ const ProductViewMissing = ({ basePath, products, t }) => {
    * @returns {Array}
    */
   const filterAvailableProducts = () => {
-    const basePathDirs = basePath.split('/');
-    const updatedProducts = {};
-
-    basePathDirs.forEach(dir => {
-      if (dir) {
-        products.forEach(({ id, productParameter, isSearchable, ...navItem }) => {
-          if (isSearchable && new RegExp(dir, 'i').test(productParameter)) {
-            updatedProducts[id] = {
-              id,
-              productParameter,
-              isSearchable,
-              ...navItem
-            };
-          }
-        });
-      }
-    });
-
-    const filteredProducts = Object.values(updatedProducts);
-    return (
-      (filteredProducts.length && filteredProducts) ||
-      routerConfig.navigation.filter(
-        ({ isSearchable, productParameter }) => (isSearchable && productParameter) || false
-      )
-    );
+    const { configs, searchable } = routerHelpers.getRouteConfigByPath();
+    return (searchable.length && searchable) || configs;
   };
 
   /**
    * On click, update history.
    *
    * @event onUpdateHistory
-   * @param {string} path
+   * @param {string} id
    * @returns {void}
    */
-  const onClick = path => history.push(path);
+  const onClick = id => {
+    const { routeHref } = routerHelpers.getRouteConfig({ id });
+    history.push(routeHref);
```

Comment:
future: being able to minimally confirm the onclick is working would help for unit testing. 

notes:
appears that emulating `react-router-dom`s `history` comes back with the appearance of some false positives, the click event "doesn't not say" it fired, tests pass, but doesn't actually push the passed `routeHref`, little deceptive. An alternative we explored... works every time by passing `useHistory` as a prop and overriding it during testing. we do something similar with translate